content/en/docs/tasks/administer-cluster/migrating-from-dockershim/find-if-you-affected.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Finding if you are affected by dockershim deprecation
+content_type: task 
+reviewers:
+- SergeyKanzhelev
+weight: 20
+---
+
+<!-- overview -->
+
+This is very unlikely that your application has a dependency on dockershim. In
+order to understand how your cluster or apps may acquire dependency on Docker as
+a container runtime, you need to understand what role dockershim plays when
+container is being scheduled on the node.
+
+## Container runtimes
+
+[Container runtime](/docs/setup/production-environment/container-runtimes/) is a software that can schedule Pods. Kubernetes is responsible for orchestration of application Pods, and uses CRI (container runtime interface) to communicate to container runtime to actually schedule these Pods and get their status.
+
+There are various container runtimes, each has special features and decide on tradeoffs they make between performance, security and functionality. Dockershim is a first container runtime used by Kubernetes. When dockershim is used as a container runtime, kubelet communicates with Docker to schedule containers and check on it’s status. Docker in its turn uses Containerd to actually schedule containers, while also making these containers visible for Docker tooling and UI.
```

Comment:
```suggestion
In its earliest releases, Kubernetes offered compatibility with just one container runtime: Docker.
Later in the Kubernetes project's history, cluster operators wanted to adopt additional container runtimes.
The CRI was designed to allow this kind of flexibility - and the kubelet began supporting CRI. However,
because Docker existed before the CRI specification was invented, the Kubernetes project created an
adapter component, “dockershim”. The dockershim adapter allows the kubelet to interact with Docker as
if Docker were a CRI compatible runtime.
```